typescript-operators/main.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable no-unused-vars */
+
+interface Customer {
+  name: string;
+  details?: {
+    address?: {
+      city: string;
+      street: string;
+      zip: string;
+    };
+    age?: number;
+  };
+}
+
+const v1 = { value: 'Something' };
+const v2 = '';
+const v3 = undefined;
+const v4 = [3, 5, 7, 9];
+const v5: Customer = {
+  name: 'Carl',
+  details: {
+    age: 82,
+  },
+};
+
+v1 && console.log('v1 is truthy');
+v2 && console.log('v2 is truthy');
+
+const config1 = v1 || 'default value';
+const config2 = v2 || 'default value';
+console.log('configs:', config1, config2);
+
+const cfg1 = v1 ?? 'default value';
+const cfg2 = v2 ?? 'default value';
+const cfg3 = v3 ?? 'default value';
+console.log('cfgs:', cfg1, cfg2, cfg3);
+
+let tern1;
+let tern2;
+v1 ? (tern1 = 'truthy') : (tern1 = 'falsy');
+v2 ? (tern2 = 'truthy') : (tern2 = 'falsy');
```

Comment:
Works, but you don't need to initialize the variables ahead of time, you can do it directly in the assignment

```
const tern1 = v1 ? 'truthy' : 'falsy'
const tern2 = v2 ? 'truthy' : 'falsy'
```